### PR TITLE
Add an option to reduce compile() console spam

### PR DESCRIPTION
**Pull request by Rocketknight1**
This is a very simple PR to add an option to reduce the console spam from our `compile()` method.

Briefly, Keras model expect you to pass a loss function to `compile()`. However, `transformers` models usually compute loss internally. The solution we used was that if the user didn't pass a `loss` argument to `compile()`, we would read the model's `loss` output and use that as the loss. This was non-standard Keras behaviour, though, so we added a warning to make sure users knew what was going on.

Now that we've been doing it for a while, though, that warning is probably just more console spam. This PR adds the option to specify `loss="auto"`, which has exactly the same behaviour but eliminates the warning. The warning also includes a line telling users that they can do that.

### CODE REVIEW

**Review by sgugger — COMMENTED** [review]
Or maybe we just make the warning an info?

**Comment by Rocketknight1** [comment]
I think we tried that, but then most users would miss it! I think we definitely want to make users aware of what's going on (because this is the one big place we diverge from the Keras training standard), but give them an option to disable the warning once they know.

**Comment by sgugger** [comment]
I think proper documentation is a better solution than throwing a warning all the time. Users are pretty mad at us with those already.

**Comment by Rocketknight1** [comment]
This could be a good opportunity to document stuff, actually! Do you prefer a sidebar tutorial, or some text in the docstring that gets added to all of our TF models?

**Comment by sgugger** [comment]
I think in the official examples where we actually use `torch.compile` is probably the best place.

**Comment by Rocketknight1** [comment]
This is TF compilation, not `torch.compile()`! All of our TF examples use it.

**Comment by Rocketknight1** [comment]
Done! I added a comment when we call `compile()` in the example scripts. I also fixed up one unnecessary use of `run_eagerly` in `run_mlm.py` - oops! 

I moved the warning message to `logging.info`, so it should be invisible to most users now, and they can still pass `loss="auto"` to disable it entirely if desired.

**Review by sgugger — COMMENTED** [review]
Those and the task examples in the doc too if you don't mind :-)

**Review by sgugger — APPROVED** [review]
Thanks!
